### ISSUE TITLE
Update to render headers inside label/legend

### ISSUE THREAD

**Issue by Mottie** [department-of-veterans-affairs/vets-design-system-documentation#2078]
# Support request

- [x] I’ve searched for any related issues/questions on github

## How can we help?

To support the minimal header pattern, we need to update `va-text-input`, `va-number-input`, `va-checkbox-group`, `va-date`, `va-memorable-date`, and `va-textarea` to render headers of a set level inside the label/legend. The `va-radio` web component already has a `label-header-level` property which performs the desired behavior. 

## What you have tried:

The mentioned web components currently support adding a label as a string only. So there is no way to work-around this limitation.

## What problem does this solve?

Once we start implementing the minimal header pattern, each form page will require a unique `H1` and many pages only ask a single question with no extra text. Placing a header inside the label/legend would solve this problem.

**Comment by humancompanion-usds** [comment]
I believe the a11y CoP is working on a proposal for how this should work. I don't think we're going to move on this until we understand from them what they want.

**Comment by rmessina1010** [comment]
Asking for clarification:  is the intent to be able to put a heading INSIDE  a label/legend as in: `<label><h3>something...</h3></label>`?  would that not break a number of semantic rules?

**Comment by Mottie** [comment]
We've already implemented this for the `va-radio` and `va-checkbox-group` inside the `legend`, e.g. `<legend><h3>Title</h3></legend>`. The `va-radio` header is already being used in the Supplemental Claim form, as suggested by Josh Kim.

The remaining requests are for adding a header inside a `label` of the `va-text-input` (PR opened), `va-number-input` (PR opened), `va-textarea` (PR opened), `va-date` & `va-memorable-date`.

The remaining web components were included in anticipation of needing an `h1` inside the legend/label once a minimal header pattern is in place. As you can see in the [design](https://www.sketch.com/s/0424c51d-7818-43ea-8821-6de91eb7ab27/a/j4oDeZa) vs [actual page](https://staging.va.gov/decision-reviews/appeals-testing/additional-info), we need the `H1` to contain the label text

|  |  |
|---|---|
| Tell us more hint text<br /><img alt="design" src="https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/136959/255cec8e-5592-4f7f-85e5-7bc04800e620" /> |  Tell us more label text <br /><img alt="staging" src="https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/136959/6fd8d5fb-4d7e-4f63-a69a-fcf456e56cfc" /> |

**Comment by Mottie** [comment]
Ok, I'll halt work on this for now.

**Comment by jeana-adhoc** [comment]
After much discussion/research, I'd like to recommend that the label for  `va-text-input`, `va-textarea`, and `va-number-input` *not* be styled as a header. The reason being that a header helps identify heirarchy of content, and if there is no content after (just a text input field) it's an orphaned heading with out content -- also, it's considered invalid html to write `<label><hx>heading here</hx></label>` A heading would have to be wrapped around a label.

These fields also don't natively need a fieldset or legend.

However, we should still allow a `va-text-input` to be referenced by an aria-labelledby that points to a header, like in this screenshot:

<img width="450" alt="image" src="https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/117098918/5268b680-eb5f-4ddd-b332-faa6a0e601d5">

Same with a text-area
<img width="450" alt="image" src="https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/117098918/852f0832-9fe4-480c-bd9d-c0f60d873307">

This also allows the `(*Required)` label to remain within the current webcomponent in these cases. 

Because `va-date` and `va-memorable-date` are constructed with existing fieldsets/legends and multiple other fields, I think it makes complete sense to allow a header to be passed in through the legend on those components.

**Comment by humancompanion-usds** [comment]
It looks to me like Robin has merged the PRs we needed and closed the rest. Thus I'm going to close this. I've [opened a new issue to document ](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2151)all of this, starting with Jeana's previous comment.
